packages/react-component-library/src/components/Button/Button.test.tsx

```diff
@@ -1,37 +1,92 @@
-      component = shallow(<Button onClick={onClick}>Hello World</Button>)
+      wrapper = render(<Button onClick={onClickSpy}>Click me</Button>)
+      button = wrapper.getByText('Click me').parentElement
     })
 
-    describe('and a user clicks on the button', () => {
-      let blurSpy: jest.SpyInstance
+    it('should style the button', () => {
+      expect(button.className).toEqual('rn-btn rn-btn--regular')
+    })
+
+    it('should default the type to "button"', () => {
+      expect(button).toHaveAttribute('type', 'button')
+    })
 
+    describe('when the button is clicked', () => {
       beforeEach(() => {
-        blurSpy = jest.fn()
-        component.simulate('click', {
-          currentTarget: {
+        fireEvent.click(button, {
+          target: {
             blur: blurSpy,
           },
         })
       })
 
-      it('should call the handler when the button is clicked', () => {
-        expect(onClick).toHaveBeenCalled()
+      it('should blur the button so it does not remain active', () => {
+        expect(blurSpy).toHaveBeenCalledTimes(1)
       })
 
-      it('should blur the button so it does not remain active', () => {
-        expect(blurSpy).toHaveBeenCalled()
+      it('should handle the click event', () => {
+        expect(onClickSpy).toHaveBeenCalledTimes(1)
       })
     })
   })
+
+  describe('when the size is specified', () => {
+    it.each`
+      size          | expected
+      ${'small'}    | ${'rn-btn rn-btn--small'}
+      ${'regular'}  | ${'rn-btn rn-btn--regular'}
+      ${'large'}    | ${'rn-btn rn-btn--large'}
+      ${'xlarge'}   | ${'rn-btn rn-btn--xlarge'}
+    `('styles the button when the size is $size', ({ size, expected }) => {
+      wrapper = render(<Button onClick={onClickSpy} size={size}>Click me</Button>)
+      button = wrapper.getByText('Click me').parentElement
+
+      expect(button.className).toEqual(expected)
+    })
+  })
+
+  describe('when the type is specified', () => {
+    it.each`
+      type        | expected
+      ${'button'} | ${'button'}
+      ${'submit'} | ${'submit'}
+    `('should set the type attribute to $type', ({ type, expected }) => {
+      wrapper = render(<Button onClick={onClickSpy} type={type}>Click me</Button>)
+      button = wrapper.getByText('Click me').parentElement
+
+      expect(button).toHaveAttribute('type', expected)
+    })
+  })
+
+  describe('when the variant is specified', () => {
+    it.each`
+      variant        | expected
+      ${'primary'}   | ${'rn-btn rn-btn--primary rn-btn--regular'}
+      ${'secondary'} | ${'rn-btn rn-btn--secondary rn-btn--regular'}
+      ${'tertiary'}  | ${'rn-btn rn-btn--tertiary rn-btn--regular'}
+    `('styles the button when the variant is $variant', ({ variant, expected }) => {
+      wrapper = render(<Button onClick={onClickSpy} variant={variant}>Click me</Button>)
+      button = wrapper.getByText('Click me').parentElement
+
+      expect(button.className).toEqual(expected)
```

Comment:
Not sure if I'd do this or use a assertion to see if the component contains a class name specified instead of looking for an exact match. This could break if the order changes or additional class names are added. This is just supposed to test that the type affects a particular class name